server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -28,14 +31,23 @@ runGQ
   -> GQLReq GQLQueryText
   -> m (HttpResponse EncJSON)
 runGQ reqId userInfo reqHdrs req = do
-  E.ExecutionCtx _ sqlGenCtx pgExecCtx planCache sc scVer _ enableAL <- ask
-  execPlan <- E.getResolvedExecPlan pgExecCtx planCache
-              userInfo sqlGenCtx enableAL sc scVer req
-  case execPlan of
-    E.GExPHasura resolvedOp ->
-      flip HttpResponse Nothing <$> runHasuraGQ reqId req userInfo resolvedOp
-    E.GExPRemote rsi opDef  ->
-      E.execRemoteGQ reqId userInfo reqHdrs req rsi opDef
+  -- The response and misc telemetry data:
+  let telemTransport = Telem.HTTP
+  (telemTimeTot, (telemCacheHit, telemLocality, ((telemTimeIO, telemQueryType), !resp))) <- withElapsedTime $ do
```

Comment:
Wouldn’t it be nice if Haskell had structurally typed records? 😭

Reply:
yeah it wasn't very satisfying to write this stuff...